pkg/chartverifier/reportbuilder_test.go

```diff
@@ -43,3 +48,53 @@ func TestSha(t *testing.T) {
 	}
 
 }
+
+func TestFilePackageDigest(t *testing.T) {
+
+	charts := []string{"checks/chart-0.1.0-v3.with-csi.tgz",
+		"checks/chart-0.1.0-v3.valid.tgz",
+		"checks/chart-0.1.0-v2.invalid.tgz",
+		"checks/chart-0.1.0-v3.without-readme.tgz",
+		"checks/chart-0.1.0-v3.with-crd.tgz"}
+
+	for _, chart := range charts {
+		cmd := exec.Command("sha256sum", chart)
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		err := cmd.Run()
+		assert.NoError(t, err, "error running sha256sum command")
+		commandResponse := strings.Split(out.String(), " ")
+		assert.Equal(t, commandResponse[0], GetPackageDigest(chart), fmt.Sprintf("%s digests did not match as expected", chart))
+	}
+
+}
+
+func TestUrlPackageDigest(t *testing.T) {
+
+	charts := []string{"https://github.com/redhat-certification/chart-verifier/blob/main/pkg/chartverifier/checks/chart-0.1.0-v3.valid.tgz?raw=true",
+		"https://github.com/openshift-helm-charts/charts/releases/download/hashicorp-vault-0.13.0/hashicorp-vault-0.13.0.tgz",
+		"https://github.com/openshift-helm-charts/charts/releases/download/hashicorp-vault-0.12.0/hashicorp-vault-0.12.0.tgz",
+		"https://github.com/IBM/charts/blob/master/repo/ibm-helm/ibm-object-storage-plugin-2.1.2.tgz?raw=true"}
+
+	for _, chart := range charts {
+
+		var out bytes.Buffer
+		cmd1 := exec.Command("curl", "-L", chart)
```

Comment:
Why depend on curl for the test? We can hardcode the digest right? simple map would do I think.

Reply:
we could but I baulked at that because I didn't want the test to start failing if the target chart changed for any reason, which is quite likely for our own test chart.

Reply:
If test chart change we change the she right, and the chart will be a diff version. Are you saying that we will repackage the same version? Why at the end is just a binary we want to test. We might change the location to test in diff circumstance but binary can be constant, in fact id does not have to be a full blown chart is could be a little one we push.

Reply:
I rather do not depend on underlying OS, that seems to be the best practice with go trying to build all in same executable. What if we run on a linux where curl is not available we can still unit test.

Reply:
Updated the test to use hard coded sha values